check_current_company_for_validation/models/check_current_company_mixin.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 Akretion (https://www.akretion.com).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class CheckCurrentCompanyMixin(models.AbstractModel):
+    _name = "check.current.company.mixin"
+    _description = "Check Current Company Mixin"
+
+    def check_current_company(self):
+        if self.env.company != self.company_id:
```

Comment:
This can fail if you have a recordset with different companies
```suggestion
        if self.filtered(lambda x: x.company_id != self.env.company)
```

or loop through all the recordset